web/src/user_group_edit.js

```diff
@@ -303,6 +304,33 @@ export function setup_group_list_tab_hash(tab_key_value) {
     }
 }
 
+function display_membership_toggle_spinner(group_row) {
+    /* Prevent sending multiple requests by removing the button class. */
+    $(group_row).find(".check").removeClass("join_leave_button");
+
+    /* Hide the tick. */
+    const $tick = $(group_row).find("svg");
+    $tick.addClass("hide");
+
+    /* Add a spinner to show the request is in process. */
+    const $spinner = $(group_row).find(".join_leave_status").expectOne();
+    $spinner.show();
+    loading.make_indicator($spinner);
+}
+
+function hide_membership_toggle_spinner(group_row) {
+    /* Re-enable the button to handle requests. */
+    $(group_row).find(".check").addClass("join_leave_button");
+
+    /* Show the tick. */
+    const $tick = $(group_row).find("svg");
+    $tick.removeClass("hide");
+
+    /* Destroy the spinner. */
+    const $spinner = $(group_row).find(".join_leave_status").expectOne();
+    loading.destroy_indicator($spinner);
+}
```

Comment:
This leads to some duplication. We can probably move these functions to `settings_components` and keep only single function.